src/main/java/seedu/address/storage/JsonAdaptedHomework.java

```diff
@@ -36,7 +36,15 @@ public String getHomeworkName() {
      * Converts this Jackson-friendly adapted tag object into the model's {@code Homework} object.
      */
     public Homework toModelType() {
-        return new Homework(homeworkName);
+        if (homeworkName.contains("[✓]")) {
+            String newHw = homeworkName.replace(" [✓]", "");
+            Homework hw = new Homework(newHw);
+            hw.markAsDone();
+            return hw;
+        } else {
+            String newHw = homeworkName.replace(" [ ]", "");
+            return new Homework(newHw);
```

Comment:
Make sure to update the json files accordingly and check that they work too

Reply:
updated json files and they do work!